lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2593,6 +2593,393 @@ class ConvertAtenGridSamplerOp : public OpConversionPattern<AtenGridSamplerOp> {
 };
 } // namespace
 
+static Value NearestInterpolate(OpBuilder &b, Location loc, Value outputSizeH,
+                                Value outputSizeW, Value input,
+                                Value inputSizeH, Value inputSizeW) {
+
+  auto inputType = input.getType().cast<RankedTensorType>();
+  auto inputRank = inputType.getRank();
+  Value cstOneFloat = b.create<arith::ConstantOp>(loc, b.getF32FloatAttr(1.0));
+
+  Value yOut = b.create<linalg::IndexOp>(loc, 2);
+  Value xOut = b.create<linalg::IndexOp>(loc, 3);
+
+  Value inputHFP = b.create<arith::SIToFPOp>(loc, b.getF32Type(), inputSizeH);
+  Value inputWFP = b.create<arith::SIToFPOp>(loc, b.getF32Type(), inputSizeW);
+  Value inputH = b.create<arith::SubFOp>(loc, inputHFP, cstOneFloat);
+  Value inputW = b.create<arith::SubFOp>(loc, inputWFP, cstOneFloat);
+
+  Value outputSizeHFP =
+      b.create<arith::SIToFPOp>(loc, b.getF32Type(), outputSizeH);
+  Value outputSizeWFP =
+      b.create<arith::SIToFPOp>(loc, b.getF32Type(), outputSizeW);
+
+  // x_original = x_resized / scale
+  Value hScale = b.create<arith::DivFOp>(loc, outputSizeHFP, inputH);
+  Value wScale = b.create<arith::DivFOp>(loc, outputSizeWFP, inputW);
+
+  Value yOutInt = b.create<arith::IndexCastOp>(loc, b.getI64Type(), yOut);
+  Value yOutFP = b.create<arith::SIToFPOp>(loc, b.getF32Type(), yOutInt);
+  Value yProj = b.create<arith::DivFOp>(loc, yOutFP, hScale);
+
+  Value xOutInt = b.create<arith::IndexCastOp>(loc, b.getI64Type(), xOut);
+  Value xOutFP = b.create<arith::SIToFPOp>(loc, b.getF32Type(), xOutInt);
+  Value xProj = b.create<arith::DivFOp>(loc, xOutFP, wScale);
+
+  Value yNearestFP = b.create<math::FloorOp>(loc, yProj);
+  Value xNearestFP = b.create<math::FloorOp>(loc, xProj);
+
+  Value yNearestInt =
+      b.create<arith::FPToSIOp>(loc, b.getI64Type(), yNearestFP);
+  Value yNearest =
+      b.create<arith::IndexCastOp>(loc, b.getIndexType(), yNearestInt);
+
+  Value xNearestInt =
+      b.create<arith::FPToSIOp>(loc, b.getI64Type(), xNearestFP);
+  Value xNearest =
+      b.create<arith::IndexCastOp>(loc, b.getIndexType(), xNearestInt);
+
+  SmallVector<Value> indices;
+  for (unsigned i = 0; i < inputRank; i++) {
+    indices.push_back(b.create<linalg::IndexOp>(loc, i));
+  }
+
+  int hDimOffset = 2;
+  indices[hDimOffset] = yNearest;
+  indices[hDimOffset + 1] = xNearest;
+  Value retVal = b.create<tensor::ExtractOp>(loc, input, indices);
+  return retVal;
+}
+
+namespace {
+class ConvertInterpolateNearestOp
+    : public OpConversionPattern<Aten__InterpolateSizeListScaleListOp> {
```

Comment:
How can we lower one single op from two different lowering paths?

Reply:
It does not lower so it sets failure() and will try the other path

Reply:
You don't need two different conversion patterns for one op. You can have one single pattern and lower it to two distinct paths based on the value of Mode.